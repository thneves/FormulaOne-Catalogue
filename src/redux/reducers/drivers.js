import {
  FETCH_DRIVERS_REQUEST, FETCH_DRIVERS_SUCCESS, FETCH_DRIVERS_FAILURE,
} from '../actions';

export const initialState = {
  loading: false,
  drivers: [],
  error: '',
};

const driversReducers = (state = initialState, action) => {
  switch (action.type) {
    case FETCH_DRIVERS_REQUEST:
      return { ...state, loading: true };
    case FETCH_DRIVERS_SUCCESS:
      return {
        ...state,
        loading: false,
        drivers: action.payload,
        error: '',
      };
    case FETCH_DRIVERS_FAILURE:
      return {
        ...state,
        loading: false,
        drivers: [],
        error: action.payload,
      };
    default:
      return state;
  }
};

export default driversReducers;
