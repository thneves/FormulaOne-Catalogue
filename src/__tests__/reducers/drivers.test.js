import drivers, { initialState } from '../../redux/reducers/drivers';

describe('Drivers Ranking Reducer', () => {
  it('should return the initial state', () => {
    expect(drivers(undefined, {})).toMatchSnapshot();
  });

  it('should handle Fetch Drivers Request', () => {
    expect(drivers(initialState,
      {
        type: 'FETCH_DRIVERS_REQUEST',
      })).toMatchSnapshot();
  });

  it('should handle Fetch Drivers Sucess', () => {
    expect(drivers(initialState,
      {
        type: 'FETCH_DRIVERS_SUCCESS',
      })).toMatchSnapshot();
  });

  it('should handle Fetch Drivers Failure', () => {
    expect(drivers(initialState,
      {
        type: 'FETCH_DRIVERS_FAILURE',
      })).toMatchSnapshot();
  });
});
