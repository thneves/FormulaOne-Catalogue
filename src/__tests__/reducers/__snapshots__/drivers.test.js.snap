// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Drivers Ranking Reducer should handle Fetch Drivers Failure 1`] = `
Object {
  "drivers": Array [],
  "error": undefined,
  "loading": false,
}
`;

exports[`Drivers Ranking Reducer should handle Fetch Drivers Request 1`] = `
Object {
  "drivers": Array [],
  "error": "",
  "loading": true,
}
`;

exports[`Drivers Ranking Reducer should handle Fetch Drivers Sucess 1`] = `
Object {
  "drivers": undefined,
  "error": "",
  "loading": false,
}
`;

exports[`Drivers Ranking Reducer should return the initial state 1`] = `
Object {
  "drivers": Array [],
  "error": "",
  "loading": false,
}
`;
